pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -269,9 +270,9 @@ public void healthcheck(@Suspended AsyncResponse asyncResponse) throws Exception
             asyncResponse.resume(new RestException(e));
             return;
         }
-        CompletableFuture<Producer<String>> producerFuture =
-            client.newProducer(Schema.STRING).topic(topic).createAsync();
-        CompletableFuture<Reader<String>> readerFuture = client.newReader(Schema.STRING)
+        CompletableFuture<Producer<byte[]>> producerFuture =
```

Comment:
nit: If you are using `SchemaDisabled`, you don't need switch to `byte[]` schema.

Reply:
Yes, I did the same thing at the beginning, but I encountered a problem. when I use String type, when I call getValue, it will be blocked. I haven't found the reason yet. it seems that the asynchronous result has not been obtained.

```
private void healthcheckReadLoop(CompletableFuture<Reader<String>> readerFuture,
                                     CompletableFuture<?> completablePromise,
                                     String messageStr) {
        readerFuture.thenAccept((reader) -> {
                CompletableFuture<Message<String>> readFuture = reader.readNextAsync()
                    .whenComplete((m, exception) -> {
                            if (exception != null) {
                                completablePromise.completeExceptionally(exception);
                            // getValue blocked
                            } else if (m.getValue().equals(messageStr)) {
                                completablePromise.complete(null);
                            } else {
                                healthcheckReadLoop(readerFuture, completablePromise, messageStr);
                            }
                    });
            });
    }
```

curl -v http://127.0.0.1:8080/admin/v2/brokers/health

```
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> GET /admin/v2/brokers/health HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 500 Internal Server Error
< Date: Mon, 04 Nov 2019 15:46:37 GMT
< broker-address: 127.0.0.1
< Content-Type: text/plain
< Content-Length: 914
< Server: Jetty(9.4.20.v20190813)
<

 --- An unexpected error occurred in the server ---

Message: Timed out reading

Stacktrace:

java.util.concurrent.TimeoutException: Timed out reading
	at org.apache.pulsar.broker.admin.impl.BrokersBase.lambda$null$4(BrokersBase.java:296)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
* Connection #0 to host 127.0.0.1 left intact
```

Reply:
This sounds like a bug to me. Have you take a look at the log to see if there is any NPE?